### DISCUSSION TITLE
如何使用arbitrary修改斐讯域名的dns解析结果

### DISCUSSION THREAD

**Discussion by liangxyz** [IrineSistiana/mosdns#291] [Q&A]
众所周知，斐讯的服务器已经关闭，很多需要联网的电子产品都处于残废状态，我现在需要修改斐讯域名的dns解析结果指向本地192.168.0.41 homeassistant服务器，我在配置文件中加入了arbitrary，导入外部rr.txt文件来arbitrary 返回斐讯域名的应答，但是在我的windows10上使用ping和nslookup都显示该域名不存在，所以请教下应该如何编写配置文件和rr.txt
我的配置文件如下：

log:
  level: info
  file: ''

plugin:

  ################# 服务器插件 ################

  # 启动服务器的插件
  - tag: main_server
    type: server
    args:
      entry:  # 这个服务器插件收到请求后会
        - arbitrary
        - main_sequence   # 运行主执行序列
        - modify_ttl      # 修改应答的 ttl

      server:   # 监听以下协议和端口
        - protocol: udp
          addr: 0.0.0.0:53
        - protocol: tcp
          addr: 0.0.0.0:53

  ################# 可执行插件 ################
  - tag: arbitrary
    type: 'arbitrary'
    args:
    rr:   # resource record
       - 'ext:/etc/mosdns/rr.txt'  # 以 `ext:` 开头，从外部文件导入。
  # 包含分流的逻辑的插件
  - tag: main_sequence
    type: sequence
    args:
      exec:
        - if:
            - query_is_ad_domain    # 已知的广告域名
          exec:
            - _block_with_nxdomain  # 生成 NXDOMAIN 应答
            - _return               # 返回。不再执行后续插件。

        - mem_cache      # 运行缓存插件。放在这里就可以避免缓存到无用的广告域名。

        - if:
            - query_is_local_domain   # 已知的本地域名
            - '!_query_is_common'     # 和不常见的请求类型
          exec:
            - forward_local           # 用本地服务器
            - _return

        - if:
            - query_is_non_local_domain  # 已知的非本地域名
          exec:
            - _prefer_ipv4               # 优先 IPv4
            - forward_remote             # 用远程服务器
            - _return

        # 剩下的未知域名用 IP 分流。原理类似 ChinaDNS，但更稳定。
        # 详细分流原理请参考 `fallback` 的工作流程。
        - primary:
            - forward_local     # 本地服务器为主。
            - if:
                - '!response_has_local_ip'  # 过滤掉非本地的 IP。
              exec:
                - _drop_response
          secondary:
            - _prefer_ipv4
            - forward_remote
          fast_fallback: 200  # 这里建议设置成 local 服务器正常延时的 2~5 倍。
                              # 这个延时保证了 local 延时偶尔变高时，其结果不会被 remote 抢答。
                              # 如果 local 超过这个延时还没响应，可以假设 local 出现了问题。
                              # 这时用就采用 remote 的应答。单位: 毫秒。
          always_standby: true

  # 缓存
  - tag: 'mem_cache'
    type: 'cache'
    args:
      redis: 'redis://redis:6379/0'
      size: 4096
      lazy_cache_ttl: 86400
      lazy_cache_reply_ttl: 30

  # 修改应答 ttl
  - tag: 'modify_ttl'
    type: 'ttl'
    args:
      minimal_ttl: 300
      maximum_ttl: 3600

  # 转发请求至本地服务器的插件
  - tag: forward_local
    type: forward
    args:
      upstream:
        - addr: 222.172.200.68
        - addr: quic://i.passcloud.xyz:784
        - addr: https://223.5.5.5/dns-query

  # 转发请求至远程服务器的插件
  - tag: forward_remote
    type: forward
    args:
      upstream:
        - addr: https://1.1.1.1/dns-query
        - addr: https://doh.sb/dns-query

  ################ 匹配器插件 #################

  # 匹配本地域名的插件
  - tag: query_is_local_domain
    type: query_matcher
    args:
      domain:
        - 'ext:./geosite.dat:cn'

  # 匹配非本地域名的插件
  - tag: query_is_non_local_domain
    type: query_matcher
    args:
      domain:
        - 'ext:./geosite.dat:geolocation-!cn'

  # 匹配广告域名的插件
  - tag: query_is_ad_domain
    type: query_matcher
    args:
      domain:
        - 'ext:./geosite.dat:category-ads-all'

  # 匹配本地 IP 的插件
  - tag: response_has_local_ip
    type: response_matcher
    args:
      ip:
        - 'ext:./geoip.dat:cn'

**Comment by liangxyz** [comment]
rr.txt的内容如下
aircat.phicomm.com IN A ANSWER aircat.phicomm.com. IN A 192.168.0.41

**Comment by IrineSistiana** [comment]
参数 rr 的缩进错了。

**Reply by liangxyz** [reply]
> 参数 rr 的缩进错了。

感谢，原来是如此低级错误
